### DISCUSSION TITLE
Best objective 0.000000000000e+00, best bound 0.000000000000e+00, gap 0.0000%

### DISCUSSION THREAD

**Discussion by wendy444444** [yalmip/YALMIP#1112] [Possible bug or missing functionality]
I used matlab+yalmipi to build a model of MILP, and the results obtained when using gurobi to solve it are as follows. 
Set parameter Username
Set parameter IntFeasTol to value 1e-06
Set parameter ScaleFlag to value 1
Set parameter BarIterLimit to value 2000000000
Set parameter CrossoverBasis to value 0
Set parameter MinRelNodes to value 0
Set parameter NodeMethod to value 1
Set parameter PumpPasses to value 0
Set parameter ZeroObjNodes to value 0
Set parameter AggFill to value 10
Set parameter PreSparsify to value 0
Set parameter TuneTrials to value 2
Academic license - for non-commercial use only - expires 2022-06-11
Gurobi Optimizer version 9.5.1 build v9.5.1rc2 (win64)
Thread count: 2 physical cores, 4 logical processors, using up to 4 threads
Optimize a model with 2171 rows, 400 columns and 4047 nonzeros
Model fingerprint: 0x13d44653
Variable types: 354 continuous, 46 integer (46 binary)
Coefficient statistics:
  Matrix range     [8e-02, 3e+03]
  Objective range  [2e-01, 9e+00]
  Bounds range     [1e+00, 1e+00]
  RHS range        [1e+00, 3e+03]
Found heuristic solution: objective 0.0000000
Presolve removed 2171 rows and 400 columns
Presolve time: 0.02s
Presolve: All rows and columns removed
Explored 0 nodes (0 simplex iterations) in 0.06 seconds (0.00 work units)
Thread count was 1 (of 4 available processors)
Solution count 1: 0 
Optimal solution found (tolerance 1.00e-04)
Best objective 0.000000000000e+00, best bound 0.000000000000e+00, gap 0.0000%

Then I used cplex to solve it, and the results are as follows. 
CPXPARAM_Simplex_Display                         2
CPXPARAM_Barrier_Display                         2
Tried aggregator 1 time.
MIP Presolve eliminated 2171 rows and 400 columns.
MIP Presolve modified 198 coefficients.
All rows and columns eliminated.
Presolve time = 0.03 sec. (0.92 ticks)
Root node processing (before b&c):
  Real time             =    0.05 sec. (1.06 ticks)
Parallel b&c, 4 threads:
  Real time             =    0.00 sec. (0.00 ticks)
  Sync time (average)   =    0.00 sec.
  Wait time (average)   =    0.00 sec.
                          ------------
Total (root+branch&cut) =    0.05 sec. (1.06 ticks)

I don't know what went wrong, please help me solve it. Thank you very much.

**Comment by johanlofberg** [comment]
You would have to tell us what you mean is wrong. Both solvers find the problem super trivial.

**Reply by johanlofberg** [reply]
You definitely don't have too many constraints as both solvers trivially solve the problem in presolve without problems (read the logs, solved in presolve eliminated all constraints and variables...)

However, your code is weird as you have various implies statements outside constraint definitions.

**Reply by wendy444444** [reply]
Sorry, I don't understand what you mean by implies statements. Do you mean calculations other than objective functions? This is part of what I want to optimize. How can I modify it?

**Reply by johanlofberg** [reply]
All constraints have to be added as, in your notation,

`C = [C, more constraints]`

**Reply by wendy444444** [reply]
Thanks, but after adding what  you  said,the indicator in the program reported an error, so I added 
`indicator=zeros(numbus,tt-1);` , 
and  then "When" varargout "is called, one or more output parameters are not assigned". 
Why is there such a problem?and  what  should  I do, please?

**Reply by johanlofberg** [reply]
I don't know what you are doing, but a better setup is to define indicator as a binary variable, and then add the constraint indicator(i,t)==Er(i,t)-Er(i,t-1). That makes all implications cleaner as they involve single binary activator variables instead of expressions

**Reply by wendy444444** [reply]
Thank you for your wise advice, but the problem remains unsolved. The following is the error message.

When" varargout "is called, one or more output parameters are not assigned
出错 implies (line 62)
        varargout{1} = implies_internal(varargin{3:end});

出错 lmi/expandmeta (line 7)
    Fnew{i} = feval(F.clauses{i}.data{1},'expand',[],F.clauses{i}.data{2:end});

出错 expandmodel (line 47)
        F = expandmeta(F);

出错 compileinterfacedata (line 116)
    [F,failure,cause,operators] = expandmodel(F,h,options);

出错 solvesdp (line 231)
[interfacedata,recoverdata,solver,diagnostic,F,Fremoved,ForiginalQuadratics] =
compileinterfacedata(F,[],logdetStruct,h,options,0,solving_parametric);

出错 optimize (line 31)
[varargout{1:nargout}] = solvesdp(varargin{:});

出错 fenqu1_1 (line 299)
optimize(C,Objective,options);
